### ISSUE TITLE
Many "Unexpected error running resulting class file for X.java" test failures in I-build on Java 22

### ISSUE THREAD

**Issue by akurtakov** [eclipse-jdt/eclipse.jdt.core#2983]
See https://download.eclipse.org/eclipse/downloads/drops4/I20240918-0950/testresults/html/org.eclipse.jdt.core.tests.compiler_ep434I-unit-cen64-gtk3-java22_linux.gtk.x86_64_22.html

**Comment by srikanth-sankaran** [comment]
@mpalat - thanks for taking a look - please reassign as suited after triage

**Comment by stephan-herrmann** [comment]
This is just the 21 variant of #2982 right?

**Comment by srikanth-sankaran** [comment]
Would the practice of making sure that the BETA_JAVA_XX branch is tested with all JREs that the github CI testing uses catch these problems ahead of the merge ?

**Comment by mpalat** [comment]
> Would the practice of making sure that the BETA_JAVA_XX branch is tested with all JREs that the github CI testing uses catch these problems ahead of the merge ?

Essentially yes,  Y builds with earlier JREs - I will create an issue from Java 24 onwards to capture this

**Comment by stephan-herrmann** [comment]
> Would the practice of making sure that the BETA_JAVA_XX branch is tested with all JREs that the github CI testing uses catch these problems ahead of the merge ?

It's all here: https://ci.eclipse.org/releng/job/YPBuilds/job/ep433Y-unit-cen64-gtk3-java21/29/testReport/

I see the problem in monitoring those test results.

I personally was alarmed (admittedly late in the cycle) by seeing hundreds of failures when running BETA_JAVA23 tests on Java 23, and strictly focused on that aspect. When finally I found out that this problem was caused by a JIT bug in Java 23 and when finally I found a way to coerce that JIT not to throw NPE from `null`-free code, I simply had no time to check tests on other Java versions.

Ergo: monitoring test results from Y-Buids must be addressed much more actively, much earlier.
